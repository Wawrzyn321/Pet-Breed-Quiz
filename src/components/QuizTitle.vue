<template id="quiz-title-template">
  <h2>Question {{ text }}</h2>
</template>

<script>
export default {
  computed: {
    text() {
      return this.questionIndex + 1 + "/" + this.questionsCount;
    }
  },
  props: {
    // zero-based!
    questionIndex: {
      required: true,
      type: Number
    },
    questionsCount: {
      required: true,
      type: Number
    }
  }
};
</script>

<style lang="scss" scoped>
h2 {
  text-align: center;
}
</style>
