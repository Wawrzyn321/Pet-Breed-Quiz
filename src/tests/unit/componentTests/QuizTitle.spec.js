import QuizTitle from "@/components/QuizTitle";
import { shallowMount } from "@vue/test-utils";

describe("QuizTitle", () => {
  it("renders with minimal props", () => {
    const wrapper = shallowMount(QuizTitle, {
      propsData: { questionsCount: 20, questionIndex: 7 }
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it("renders title when passed question count and current index ", () => {
    const title = "Question 4/10";
    const wrapper = shallowMount(QuizTitle, {
      propsData: { questionsCount: 10, questionIndex: 3 }
    });
    expect(wrapper.text()).toBe(title);
  });
});
