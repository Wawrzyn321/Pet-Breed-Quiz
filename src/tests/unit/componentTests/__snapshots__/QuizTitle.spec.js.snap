// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizTitle renders with minimal props 1`] = `
<h2>
  Question 8/20
</h2>
`;
